app/src/main/java/com/hifeful/mealmania/presentation/myMeals/MyMealsFragmentBinder.kt

```diff
@@ -4,26 +4,19 @@ import androidx.recyclerview.widget.RecyclerView
 import com.hifeful.mealmania.databinding.FragmentMyMealsBinding
 
 fun FragmentMyMealsBinding.bind(
-    viewState: MyMealsViewState,
+    viewState: MyMealsState,
     recentMealsAdapter: RecentMealsAdapter?,
     favouriteMealsAdapter: FavouriteMealsAdapter?
 ) {
     recentMealsAdapter?.submitList(viewState.recentMeals)
 
     favouriteMealsAdapter?.apply {
-        val adapterDataObserver = object : RecyclerView.AdapterDataObserver() {
-
-            override fun onItemRangeChanged(positionStart: Int, itemCount: Int) {
-                recyclerViewFavouriteMeals.smoothScrollToPosition(0)
-            }
-
+        registerAdapterDataObserver(object : RecyclerView.AdapterDataObserver() {
```

Comment:
I would place it in the adapter declaration, to avoid registration each time while items are updated

Reply:
Good catch. Fixed